File: 1-two-sum/two-sum.cpp
class Solution {
public:
    std::vector<int> twoSum(const std::vector<int>& nums, int target) {
    std::unordered_map<int, int> hash_table;

    for (int i = 0; i < nums.size(); ++i) 
    {
        int complement = target - nums[i];

        if (hash_table.find(complement) != hash_table.end())
            return {hash_table[complement], i};
        hash_table[nums[i]] = i;
    }

    return{};
}
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mymap = {}
        for i in range(len(nums)):
            if target - nums[i] in mymap:
                return [i, mymap[target-nums[i]]]
            mymap[nums[i]] =  i
        return [-1, -1]
